quickwit/quickwit-indexing/src/source/pulsar_source.rs

```diff
@@ -376,7 +376,16 @@ async fn create_pulsar_consumer(
 fn msg_id_to_position(msg: &MessageIdData) -> Position {
     // The order of these fields are important as they affect the sorting
     // of the checkpoint positions.
-    // TODO: Confirm this layout is correct?
+    //
+    // The key parts of the ID used for ordering are:
+    // - The ledger ID which is a sequentially increasing ID.
```

Comment:
To be confirmed, which unfortunately just involves a lot of source code reading :(